src/mito2/src/manifest/storage.rs

```diff
@@ -134,6 +147,10 @@ pub struct ManifestObjectStore {
     object_store: ObjectStore,
     compress_type: CompressionType,
     path: String,
+    /// Stores the size of each manifest file.
+    /// K is the manifest file name(such as "0000000000000000006.checkpoint"
+    /// or "0000000000000000006.json"), and V is the file size.
+    manifest_size_map: HashMap<String, u64>,
```

Comment:
nitpick: Could we define an enum for the key? It should be more space-efficient than string.
```rust
/// Key to identify a manifest file.
enum FileKey {
    /// A delta file (`.json`).
    Delta(ManifestVersion),
    /// A checkpoint file (`.checkpoint`).
    Checkpoint(ManifestVersion),
}
```